kratos/utilities/properties_accessor.h

```diff
@@ -0,0 +1,327 @@
+                                                const GeometryType::Pointer&,
+                                                const DataValueContainer&,
+                                                const ProcessInfo&,
+                                                const Vector&,
+                                                const int)>;
+
+    using DoubleConfiguration = std::map<DoubleVariableType, DoubleFunction>;
+
+    // std::function<void(int)>
+
+    // using std::unordered_map<DoubleVariableType>
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    PropertiesAccessor(IndexType NewId = 0) : BaseType(NewId) {}
+
+    /// Destructor.
+    virtual ~PropertiesAccessor() {}
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    template<class TVariableType>
+    typename TVariableType::Type& GetValue(const TVariableType& rV,
+                                           const Properties::Pointer& rpProperties,
+                                           const GeometryType::Pointer& rpGeometry,
+                                           const DataValueContainer& rDataContainer,
+                                           const ProcessInfo& rCurrentProcessInfo,
+                                           const Vector& rShapeFunctionValues,
+                                           const int GaussPointIndex=0)
+    {
+        // rpProperties->GetValue(rV);
+    }
+
+
+    double GetValue(const DoubleVariableType& rV,
+                     const Properties::Pointer& rpProperties,
+                     const GeometryType::Pointer& rpGeometry,
+                     const DataValueContainer& rDataContainer,
+                     const ProcessInfo& rCurrentProcessInfo,
+                     const Vector& rShapeFunctionValues,
+                     const int GaussPointIndex=0)
+    {
+        if(mConfigurationsDouble.find(rV) != mConfigurationsDouble.end())
+            return mConfigurationsDouble[rV](rpProperties,
+                                             rpGeometry,
+                                             rDataContainer,
+                                             rCurrentProcessInfo,
+                                             rShapeFunctionValues,
+                                             GaussPointIndex);
+        else
+            return rpProperties->GetValue(rV);
+    }
+
+    void Configure(const DoubleVariableType& rV, std::function<double(const Properties::Pointer&,
+                                                const GeometryType::Pointer&,
+                                                const DataValueContainer&,
+                                                const ProcessInfo&,
+                                                const Vector&,
+                                                const int)> DoubleFunction)
+    {
+        mConfigurationsDouble[rV] = DoubleFunction;
```

Comment:
as a proof of concept it looks good to me...however i think that the syntax is not correct :-)

Reply:
Yep here I was lazy, will make it proper ofc